### PR TITLE
for checking

### PR DESCRIPTION
**Pull request by TatsianaKryshtofik**
Check please my models

### CODE REVIEW

File: blog/managers.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from django.contrib.auth.base_user import BaseUserManager
+
+class UserManager(BaseUserManager):
+    use_in_migrations = True
+
+
+    def _create_user(self, email, password, **extra_fields):
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
```
Comment (MaximShidlovski23) [inline blog/managers.py:9]: think about using normalize_email()

File: blog/admin.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+admin.site.register(User)
+admin.site.register(UserInfo)
+admin.site.register(Post)
+admin.site.register(Category)
+admin.site.register(Subcategory)
+admin.site.register(Comment)
+admin.site.register(Tag)
+admin.site.register(PostRating)
+admin.site.register(Image)
```
Comment (MaximShidlovski23) [inline blog/admin.py:12]: each file with your code should end with a newline

File: blog/models.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
+from django.db import models
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.utils.translation import ugettext_lazy as _
+
+
+from .managers import UserManager
+
+
+class User(AbstractBaseUser):
+
```
Comment (MaximShidlovski23) [inline blog/models.py:9]: Use doc string

File: blog/models.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    class Meta:
+        verbose_name = _('user')
+        verbose_name_plural = _('users')
+
+    def get_full_name(self):
+        full_name = '{} {}'.format(self.first_name, self.last_name)
+        return full_name.strip()
+
+    def get_short_name(self):
```
Comment (MaximShidlovski23) [inline blog/models.py]: why do you need this method?

File: blog/managers.py (hunk 1 above)
Comment (MaximShidlovski23) [inline blog/managers.py:8]: think about using normalize_email()

File: blog/admin.py (hunk 2 above)
Comment (MaximShidlovski23) [inline blog/admin.py]: each file with your code should end with a newline

File: blog/managers.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, email, password=None, **extra_fields):
+        extra_fields.setdefault('is_superuser', False)
+        return self._create_user(email, password, **extra_fields)
+
+    def create_superuser(self, email, password, **extra_fields):
+        extra_fields.setdefault('is_superuser', True)
```
Comment (MaximShidlovski23) [inline blog/managers.py:18]: Maybe you should add the is_admin=True flag

File: blog/managers.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+from django.contrib.auth.base_user import BaseUserManager
+
+class UserManager(BaseUserManager):
+    use_in_migrations = True
+
+
+    def _create_user(self, email, password, **extra_fields):
```
Comment (MaximShidlovski23) [inline blog/managers.py:7]: user must have a username

File: blog/models.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.utils.translation import ugettext_lazy as _
+
+
+from .managers import UserManager
+
+
+class User(AbstractBaseUser):
+
+    email = models.EmailField(_('email'), unique=True)
```
Comment (MaximShidlovski23) [inline blog/models.py]: add username

File: blog/models.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
+from django.db import models
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.utils.translation import ugettext_lazy as _
```
Comment (MaximShidlovski23) [inline blog/models.py]: Why do you use ugettext_lazy?

File: blog/models.py [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+    last_name = models.CharField(_('surname'), max_length=50, blank=True)
+    birthday = models.DateField(_('birthday'), null=True)
+    phone_number = models.CharField(_('phone_number'), max_length=12)
+    date_joined = models.DateTimeField(_('registered'), auto_now_add=True)
+    is_active = models.BooleanField(_('is_active'), default=True)
+    avatar = models.ForeignKey('Image', on_delete=models.CASCADE)
+
+    objects = UserManager()
+
+    USERNAME_FIELD = 'email'
```
Comment (MaximShidlovski23) [inline blog/models.py]: Then change USERNAME_FIELD to username field

File: blog/models.py [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+    REQUIRED_FIELDS = []
```
Comment (MaximShidlovski23) [inline blog/models.py]: add required fields for registration

File: blog/models.py [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+    def get_full_name(self):
+        full_name = '{} {}'.format(self.first_name, self.last_name)
+        return full_name.strip()
+
+    def get_short_name(self):
+        return self.first_name
+
+
+class UserInfo(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE, primary_key=True)
```
Comment (MaximShidlovski23) [inline blog/models.py]: maybe it would be better to use the foreign key from user

File: blog/models.py [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+    country = models.CharField(_('country'), max_length=50)
+    city = models.CharField(_('city'), max_length=50)
+    address = models.CharField(_('address'), max_length=50)
+    phone = models.CharField(_('phone'), max_length=12)
```
Comment (MaximShidlovski23) [inline blog/models.py]: you have a phone_number in User, decide where you really need it

File: blog/models.py [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+        verbose_name_plural = _('users info')
+
+
+class Post(models.Model):
+
+    user = models.ForeignKey(User, verbose_name=_('author'), on_delete=models.CASCADE)
+    category = models.ForeignKey('category', on_delete=models.SET_NULL, null=True)
+    subcategory = models.ForeignKey('subcategory', on_delete=models.SET_NULL, null=True)
+    title = models.CharField(_('title'), max_length=50)
+    body = models.TextField('text', blank=True)
```
Comment (MaximShidlovski23) [inline blog/models.py]: Maybe it's better to call this field description

File: blog/models.py [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+        ordering = ['created', 'title']
+
+    def __str__(self):
+        return self.title
+
+
+class Category(models.Model):
+
+    title = models.CharField(_('title'), max_length=50)
+    subtitle = models.ManyToManyField('subcategory')
```
Comment (MaximShidlovski23) [inline blog/models.py:72]: You don't have description

File: blog/models.py [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+        verbose_name = _('category')
+        verbose_name_plural = _('categories')
+
+    def __str__(self):
+        return self.title
+
+
+class Subcategory(models.Model):
+    title = models.CharField(_('title'), max_length=50)
+    created = models.DateTimeField(_('updated'), auto_now_add=True)
```
Comment (MaximShidlovski23) [inline blog/models.py]: You don't have description

File: blog/models.py [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+        return self.title
+
+
+class Comment(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    post = models.ForeignKey(Post, on_delete=models.CASCADE)
+    body = models.TextField('text')
+    created = models.DateTimeField(_('created'), auto_now_add=True)
+    updated = models.DateTimeField(_('updated'), auto_now=True)
+
```
Comment (MaximShidlovski23) [inline blog/models.py:115]: maybe you need the deleted field?

File: blog/models.py [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+    class Meta:
+        verbose_name = _('comment')
+        verbose_name_plural = _('comments')
+
+
+class PostRating(models.Model):
+
+    value = models.SmallIntegerField(_('value'))
+    created = models.DateTimeField(_('created'), auto_now_add=True)
+    update = models.DateTimeField(_('updated'), auto_now=True)
```
Comment (MaximShidlovski23) [inline blog/models.py]: updated

File: blog/models.py [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+        return self.value
+
+    class Meta:
+        verbose_name = _('rating')
+        verbose_name_plural = _('ratings')
+
+
+class Image(models.Model):
+
+    image_url = models.URLField(_('image_url'), max_length=50)
```
Comment (MaximShidlovski23) [inline blog/models.py]: increase the max_length to 150

File: blog/models.py [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+    length = models.CharField(_('length'), max_length=50)
```
Comment (MaximShidlovski23) [inline blog/models.py]: length and width must be a number

File: numero/settings.py [hunk 20]
```diff
@@ -0,0 +1,131 @@
...
+WSGI_APPLICATION = 'numero.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/3.2/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'numero_db',
```
Comment (MaximShidlovski23) [inline numero/settings.py]: transfer all user data to config file

**Review by MaximShidlovski23 — CHANGES_REQUESTED** [review]
You need to fix something

File: blog/admin.py [hunk 21]
```diff
@@ -9,4 +9,4 @@
 admin.site.register(Comment)
 admin.site.register(Tag)
 admin.site.register(PostRating)
-admin.site.register(Image)
\ No newline at end of file
+admin.site.register(Image)
```
Comment (MaximShidlovski23) [inline blog/admin.py:12]: Have you fixed it everywhere?

File: blog/models.py [hunk 22]
```diff
@@ -19,7 +19,7 @@ class User(AbstractBaseUser):
     objects = UserManager()
 
     USERNAME_FIELD = 'email'
-    REQUIRED_FIELDS = []
+    REQUIRED_FIELDS = ['email', 'password', 'username']
```
Comment (MaximShidlovski23) [inline blog/models.py]: Yesterday I said that you only need to add username and email

File: blog/managers.py [hunk 23]
```diff
@@ -15,9 +15,9 @@ def create_user(self, email, password=None, **extra_fields):
         return self._create_user(email, password, **extra_fields)
 
     def create_superuser(self, email, password, **extra_fields):
-        extra_fields.setdefault('is_superuser', True)
+        extra_fields.setdefault('is_admin', True)
```
Comment (MaximShidlovski23) [inline blog/managers.py:20]: Yesterday we discussed this, you need to use 2 flags, Not only is_admin, but also Is_superuser

**Review by MaximShidlovski23 — CHANGES_REQUESTED** [review]

File: blog/managers.py (hunk 23 above)
Comment (TatsianaKryshtofik) [reply blog/managers.py:20]: I deleted by accident

File: blog/admin.py (hunk 21 above)
Comment (TatsianaKryshtofik) [reply blog/admin.py:12]: Yes, I have. I`ve checked

File: blog/models.py (hunk 22 above)
Comment (TatsianaKryshtofik) [reply blog/models.py]: Done

File: blog/models.py [hunk 24]
```diff
@@ -108,7 +108,7 @@ def __str__(self):
 class Comment(models.Model):
     user = models.ForeignKey(User, on_delete=models.CASCADE)
     post = models.ForeignKey(Post, on_delete=models.CASCADE)
-    description = models.TextField('text')
+    description = models.TextField('text', blank=True)
     created_at = models.DateTimeField('created_at', auto_now_add=True)
     updated_at = models.DateTimeField('updated_at', auto_now=True)
```
Comment (MaximShidlovski23) [inline blog/models.py:115]: you didn't add the deleted_at field in the comments

**Review by MaximShidlovski23 — CHANGES_REQUESTED** [review]

File: blog/models.py [hunk 25]
```diff
@@ -111,6 +111,7 @@ class Comment(models.Model):
     description = models.TextField('text', blank=True)
     created_at = models.DateTimeField('created_at', auto_now_add=True)
     updated_at = models.DateTimeField('updated_at', auto_now=True)
+    deleted_at = models.DateTimeField('deleted_at', auto_now=True)
```
Comment (MaximShidlovski23) [inline blog/models.py]: deleted_at mustn't auto_now flag, it need only null=true

**Review by MaximShidlovski23 — CHANGES_REQUESTED** [review]

**Pull request merged**